popoolation2.rb

```diff
@@ -0,0 +1,101 @@
+        DNA (Pool-Seq). It may be used for measuring differentiation between populations, for genome wide
+        association studies and for experimental evolution."
+  homepage "https://sourceforge.net/p/popoolation2/wiki/Main/"
+  url "https://downloads.sourceforge.net/project/popoolation2/popoolation2_1201.zip"
+  version "1.2.01"
+  sha256 "7f45353a211ec88d36fd1beed1d128673f3ff911b731211eb5a39f8602bddd42"
+
+  bottle :unneeded
+
+  depends_on "cpanminus" => :build
+  depends_on :perl => ["5.8", :run]
+  depends_on :java => ["1.8", :run, :recommended]
+  depends_on "r" => [:run, :optional]
+  depends_on "samtools" => [:run, :optional]
+  depends_on "bwa" => [:run, :optional]
+
+  def install
+    rm "mpileup2sync.pl"
+    chmod 0755, Dir.glob("*.pl")
+    chmod 0755, Dir.glob("export/*.pl")
+    chmod 0755, Dir.glob("indel_filtering/*.pl")
+    mkdir "foo"
+    mv Dir.glob("*.pl"), "foo/"
+    mv Dir.glob("*.jar"), "foo/"
+    mkdir "foo/export"
+    mv Dir["export/*.pl"], "foo/export/"
+    mkdir "foo/indel_filtering"
+    mv Dir["indel_filtering/*.pl"], "foo/indel_filtering/"
+    mkdir "foo/Modules"
+    mv Dir["Modules/*.pm"], "foo/Modules/"
+    mkdir "foo/Modules/Test"
+    mv Dir["Modules/Test/*.pm"], "foo/Modules/Test/"
+    mkdir "bar"
+    system "cpanm", "-L", "bar", "Text::NSP::Measures::2D::Fisher::twotailed"
+    mv "bar/lib/perl5/Text", "foo/Modules/"
+    File.write("ppool2", <<-EOS.undent)
+      \#!/bin/bash
+      atom="$1"
+      if [ "x$atom" != "x" ]; then
+        shift
+        if [ "$atom" = "mpileup2sync" ]; then
+          exec java ${JAVA_OPTS} -jar #{pkgshare}/mpileup2sync.jar "$@"
+        else
+          case "${atom}" in
+            "cmh-test"|"create-genewise-sync"|"fisher-test"|"fst-sliding"|"mpileup2sync"|\\
+            "snp-frequency-diff"|"subsample-synchronized"|"synchronize-pileup"|\\
+            "export/cmh2gwas"|"export/pwc2igv"|"export/subsample_sync2GenePop"|"export/subsample_sync2fasta"|\\
+            "indel_filtering/filter-sync-by-gtf"|"indel_filtering/identify-indel-regions")
+              script=#{pkgshare}/${atom}.pl
+              exec ${script} "$@"
+              ;;
+          esac
+        fi
+      fi
+      echo "Usage: $0 <command> <arguments>
+
+            where <command> is one of
+
+              help
+              cmh-test
+              create-genewise-sync
+              fisher-test
+              fst-sliding
+              mpileup2sync
+              snp-frequency-diff
+              subsample-synchronized
+              synchronize-pileup
+              export/cmh2gwas
+              export/pwc2igv
+              export/subsample_sync2GenePop
+              export/subsample_sync2fasta
+              indel_filtering/filter-sync-by-gtf
+              indel_filtering/identify-indel-regions
+
+           See https://sourceforge.net/p/popoolation2/wiki/Main/ for details."
+    EOS
+    bar=share/"popoolation2"
```

Comment:
Meaningful variable name. I assume while looking at the next line that the `bar` is the folder for the script dependencies. In that case, I will make the folder `share/.popoolation2`; the dot for hiding it from ls and also to have a different name as the script (if you change it for "popoolation2") and to make clear that it is related directly with the launch-script.

Reply:
I think having a share subfolder with the name of the package that installed the stuff is the normal way to do it.

Reply:
True, but I would not like that it is accessible. Anyway, it is alright without the dot, but with the same name as the script if possible.